src/cozy/wrappers/icon-sprite/icon-sprite.component.ts

```diff
@@ -0,0 +1,68 @@
+import {
+    AfterViewInit,
+    Component,
+    OnChanges,
+    OnDestroy,
+    OnInit,
+    ViewEncapsulation,
+} from '@angular/core';
+// @ts-ignore
+import { Sprite as IconSprite } from 'cozy-ui/transpiled/react/Icon';
+import * as invariant from 'invariant';
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+import * as uuid from 'uuid';
+@Component({
+    selector: 'app-icon-sprite',
+    templateUrl: './icon-sprite.component.html',
+    encapsulation: ViewEncapsulation.None,
+})
+export class IconSpriteComponent
+    implements OnChanges, OnDestroy, OnInit, AfterViewInit
+{
+    rootDomID: string = '';
+
+    /*************/
+    /* Lifecycle */
+    /*************/
+
+    ngOnInit() {
+        this.rootDomID = uuid.v1();
+    }
+
+    ngOnChanges() {
+        this.renderReact();
+    }
+
+    ngAfterViewInit() {
+        this.renderReact();
+    }
+
+    ngOnDestroy() {
+        // Uncomment if Angular 4 issue that ngOnDestroy is called AFTER DOM node removal is resolved
+        // ReactDOM.unmountComponentAtNode(this.getRootDomNode())
+    }
+
+    /**********/
+    /* Render */
+    /**********/
+
+    protected getRootDomNode() {
+        const node = document.getElementById(this.rootDomID);
+        invariant(node, `Node '${this.rootDomID} not found!`);
+        return node;
+    }
+
+    protected renderReact() {
+        if (this.isMounted()) {
+            ReactDOM.render(
+                React.createElement(IconSprite),
+                this.getRootDomNode()
+            );
+        }
+    }
+
+    private isMounted(): boolean {
+        return !!this.rootDomID;
+    }
+}

```

Comment:
can't you use the sprite directly https://unpkg.com/browse/cozy-ui@50.9.3/transpiled/react/Icon/icons-sprite.js / https://github.com/cozy/cozy-ui/blob/master/react/Icon/Sprite.jsx#L1 

without using the react component ?

Reply:
It should be possible in theory, but Angular has some protection against imported string and I should sanitize it before using it. Without it angular imports only raw strings and exclude the svg code
![image](https://user-images.githubusercontent.com/1884255/123444500-3cd1cf00-d5d7-11eb-9bb1-62b748fa6558.png)

I can search how to implement this sanitize function if you want to.

Reply:
Nein, thk